### DISCUSSION TITLE
await login in grpc throwing error tracebacks

### DISCUSSION THREAD

**Discussion by utkarshpandey12** [speechly/speechly#188] [Q&A] [answered]
I am trying to create a transciber API endpoints using speechly grpc python example. When I test this code in my local ,machine it works fine but as I deplot this it throws these errors
Its a simple flask app hosted on pythonAnywhere
I have attached error traceback from error logs
Please help me out to fix this 
![Screenshot from 2022-05-16 17-44-54](https://user-images.githubusercontent.com/38834517/168590310-1b9f482a-3417-479e-9763-9b6aa0326f53.png)

**Comment by langma** [comment, marked as answer]
Thanks for trying out!

I think pythonanywhere is [limiting the access to external APIs](https://www.pythonanywhere.com/forums/topic/13556/), meaning that the call to Speechly API is blocked.

We will try to get the Speechly API whitelisted in pythonanywhere, hopefully this is possible in the future.

**Reply by utkarshpandey12** [reply]
@langma Can you suggest some deployment methods where Speechly services can be deployed over flask apps ?
Or maybe I should open a request to whitelist speechly on pythonanywhere!!!

**Reply by langma** [reply]
Good question, I was just searching for that... I'd suggest [Digital Ocean](https://try.digitalocean.com/deploy-python/), where you should be able to get free credits for running a service. Also [Vultr](https://www.vultr.com/vultr-vs-digitalocean/) provides free credits for new signups, so you could try that also.

And I've now submitted a request to add the Speechly API in pythonanywhere, I hope they can do it quickly :)

**Reply by utkarshpandey12** [reply]
@langma 
Does this mean I can use all the functions defined in python client or is there some catch considering I am using their free acoount ?
![Screenshot from 2022-05-17 13-28-00](https://user-images.githubusercontent.com/38834517/168760301-84140ab6-b493-4128-9144-41ac757e4fae.png)
.

**Reply by langma** [reply]
I'm afraid you have to try, I'm not sure what that means exactly. Does it mean that you can expose only http endpoints from your deployments, or that they allow calling only port 80 of api.speechly.com? I would understand the former, as certificates and domains are an additional step that they could charge for.

**Reply by utkarshpandey12** [reply]
@langma I guess its only allowing port 80 of api.speechly.com. As I tested it after its whitelisted and it still throws the error which I mentioned at the start of the thread.

**Reply by utkarshpandey12** [reply]
@langma I deployed it on heroku and it works there 

Thanks

**Comment by utkarshpandey12** [comment]
Anyone looking to host speechly service as a flask app can refer to this code : **https://github.com/utkarshpandey12/heroku-basic-flask**
